components/patient-data/ui/src/main/resources/PhenoTips/GeneClass.xml

```diff
@@ -98,7 +98,67 @@
     </gene>
     <status>
       <cache>0</cache>
-      <customDisplay/>
+      <customDisplay>{{velocity output=false}}
+#set($sP = $object.xWikiClass.get($name))
+#set ($ALL_VALUES = $sP.listValues)
+#set ($LABEL_MAP = $sP.mapValues)
+#set ($GENE_STATUS_PARENTS = {})
+#set ($GENE_STATUS_FLATLIST = [])
+
+#macro (getFlatCategoriesList)
+    #set ($levels = { "" : 0 })
+    #foreach ($s in $ALL_VALUES)
+      #set ($level = $mathtool.add($levels.get("$!GENE_STATUS_PARENTS.get($s)"), 1))
+      #set ($discard = $levels.put($s, $level))
+      #set ($item = {
+        "id" : $s,
+        "name" : "$!LABEL_MAP.get($s).value",
+        "categoryId" : "$!GENE_STATUS_PARENTS.get($s)",
+        "level" : $level
+      })
+      #set ($discard = $GENE_STATUS_FLATLIST.add($item))
+    #end
+#end
+
+#macro (extractStatusSubtypes)
+    #set ($GENE_STATUS_SUBTYPES = {})
+    #foreach ($v in $ALL_VALUES)
+      #set ($descendents = [])
+      #foreach ($v1 in $ALL_VALUES)
+        #if ($v1.startsWith("${v}_"))
+          #set ($discard = $descendents.add($v1))
+        #end
+      #end
+      #set ($discard = $GENE_STATUS_SUBTYPES.put($v, $descendents))
+    #end 
+    #foreach ($s in $GENE_STATUS_SUBTYPES.keySet())
+      #foreach ($ss in $GENE_STATUS_SUBTYPES.get($s))
+        #set ($discard = $GENE_STATUS_PARENTS.put($ss, $s))
+      #end
+    #end
+    #getFlatCategoriesList()
+#end
+
+#macro (displayOption $i)
+#foreach ($l in [1..$i.level])*#end {{html clean="false"}}&lt;label&gt;&lt;input type="radio" name="$prefix$name" id="$prefix$name$i.id" value="$i.id" #if ($i.id == $value) checked="checked" #end/>$i.name&lt;/label&gt;{{/html}}
+#end
+{{/velocity}}
+
+{{velocity}}
+#extractStatusSubtypes()
+##
+#if ($xcontext.action == 'edit')
+  #if ("$!value" == '')
+    #set ($value = $ALL_VALUES[0])
+  #end
+  #foreach ($i in $GENE_STATUS_FLATLIST)
+    #displayOption($i)
+  #end
+  {{html clean="false"}}&lt;input type="hidden" name="$prefix$name" id="${prefix}${name}" value="" /&gt;{{/html}}
+#else
+$!LABEL_MAP.get($!value).value
+#end
```

Comment:
This is the draft UI, not the tree drop-down we discussed. It was supposed to look something like this: https://js.devexpress.com/Demos/WidgetsGallery/Demo/DropDownBox/SingleSelection/jQuery/Light/ . Unfortunately this detail of the discussion did not make it into the jira issue, only on slack. Mea culpa, I'll update PT-3936.

Reply:
It was discussed today to try to reuse the style and JS code that allows to expand/collapse the entries for Age of onset or Mode of inheritance.

UI and UX:
* the 6 top level options would be shown by default;
* expandable entries (in this case, Candidate and UMC) will have an arrow pointing right (►) in front of the radio button and label, to mark the fact that they are indeed expandable
* subtrees for Candidate and UMC will be collapsed by default unless a value in that branch is already selected, in which case the branch is displayed as expanded up to that entry so that the selected value will be visible
* expanded branches will have an arrow pointing down (▼) in front of the radio button and label
* when clicking on an entry, in addition to selecting that entry the branch will automatically expand